lib/sandbox/sandbox_linux.go

```diff
@@ -172,3 +174,111 @@ func hostNetNsPath() (string, error) {
 	defer netNS.Close()
 	return netNS.Path(), nil
 }
+
+// NewNS creates a new persistent (bind-mounted) network namespace and returns
+// an object representing that namespace, without switching to it.
+func NewNS() (ns.NetNS, error) {
```

Comment:
@giuseppe Can you review this?

Reply:
the code from vendor/github.com/containernetworking/plugins/pkg/ns/ns_linux.go is much simpler than this replacement.  @saschagrunert why do we need this additional steps?  For example, why do we need `/var/run/netns` to be a mount point?

Reply:
well the code is correct, as it is the same as https://github.com/containernetworking/plugins/blob/a0eac8d7d990adc7621e6ac0adf8b351e6e6c34b/pkg/testutils/netns_linux.go#L34-L132

I am just worried that if there are multiple users of this function (e.g. we will start using it in podman), there is a potential race condition where we could recreate the `MS_SHARED|MS_REC` multiple times and all the existing  mount points will be carried over multiple times.

Reply:
Yes I know, I took the code from the new implementation which seemed cleaner to me: https://github.com/containernetworking/plugins/blob/a0eac8d7d990adc7621e6ac0adf8b351e6e6c34b/pkg/testutils/netns_linux.go#L34-L132

I can also simply choose the old one if we prefer that.

Reply:
yes please, I think it is safer for now

Reply:
Changed the implementation back, there is one open discussion point (below).